camel/functions/search_functions.py

```diff
@@ -0,0 +1,107 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from typing import List
+
+import requests
+from bs4 import BeautifulSoup
+
+from .openai_function import OpenAIFunction
+
+
+def clean_str(p: str) -> str:
+    r"""Cleans the input string by encoding and decoding it multiple times
+    to ensure it can be properly read and processed by Python code
+
+    Args:
+        p (str): The input string to be cleaned, typically from the webpage
+
+    Returns:
+        str: The cleaned string
+    """
+    return p.encode().decode("unicode-"
+                             "escape").encode("latin1").decode("utf-8")
+
+
+def get_page_obs(page: str) -> str:
```

Comment:
The function name can be improved.

Reply:
I changed it to `get_page_abstract`. Is this better?